src/js/node/http.ts

```diff
@@ -649,52 +652,43 @@ export class IncomingMessage extends Readable {
     callback();
   }
 
-  #closeBodyStream() {
+  #abortBodyStream() {
     debug("closeBodyStream()");
     var bodyStream = this.#bodyStream;
-    if (bodyStream == null) return;
+    if (!bodyStream) return;
+    bodyStream.cancel();
     this.complete = true;
     this.#bodyStream = undefined;
     this.push(null);
-    // process.nextTick(destroyBodyStreamNT, bodyStream);
   }
 
   _read(size) {
     if (this.#noBody) {
       this.push(null);
       this.complete = true;
     } else if (this.#bodyStream == null) {
-      const contentLength = this.#req.headers.get("content-length");
-      let remaining = contentLength ? parseInt(contentLength, 10) : 0;
-      this.#bodyStream = Readable.fromWeb(this.#req.body, {
-        highWaterMark: Number.isFinite(remaining) ? Math.min(remaining, 16384) : 16384,
-      });
-
-      const isBodySizeKnown = remaining > 0 && Number.isSafeInteger(remaining);
-
-      if (isBodySizeKnown) {
-        this.#bodyStream.on("data", chunk => {
-          debug("body size known", remaining);
-          this.push(chunk);
-          // when we are streaming a known body size, automatically close the stream when we have read enough
-          remaining -= chunk?.byteLength ?? 0;
-          if (remaining <= 0) {
-            this.#closeBodyStream();
-          }
-        });
-      } else {
-        this.#bodyStream.on("data", chunk => {
-          this.push(chunk);
-        });
+      const reader = this.#req.body?.getReader() as ReadableStreamDefaultReader;
+      if (!reader) {
+        this.push(null);
+        this.complete = true;
+        return;
       }
 
-      // this can be closed by the time we get here if enough data was synchronously available
-      this.#bodyStream &&
-        this.#bodyStream.on("end", () => {
-          this.#closeBodyStream();
-        });
-    } else {
-      // this.#bodyStream.read(size);
+      (async () => {
```

Comment:
we should move this to an internal function so that we avoid creating a new function each time read is called

Reply:
`#consumeReader(reader) { ...` or something